microprofile/WFLY-12609_MicroProfile_JWT_1_1.adoc

```diff
@@ -0,0 +1,109 @@
+
+At the outset activation will only occur in the following circumstances: -
+
+ * A `@org.eclipse.microprofile.auth.LoginConfig` annotation is detected with an `auth-method` of `MP-JWT`.
+ * An auth method of `MP-JWT` is specified in any web.xml.
+
+This activation scope is deliberately constrained to allow us to incrementally increase the scope at a later point. 
+
+At the point activation occurs the deployment will not make use of a configured `SecurityDomain` instead a "Virtual" `SecurityDomain` will be created to meet the requirements of the deployment and will be used across all sub-deployments.  The common `SecurityDomain` will allow for identity propagation across all sub-deployments whilst the virtual nature of the domain will make it unavailable for propagation across unrelated deployments allowing for deployment level isolation.
+
+Activation will add module dependencies as required on SmallRye JWT, Elytron JWT, and the MicroProfile JWT API.
+
+Activation will also ensure activation of the SmallRye JWT CDI extension as well as any Elytron JWT CDI extension if deemed required and any Elytron specific overrides.
+
+Finally activation will ensure the activation of the `MP-JWT` authentication mechanism for the deployment.
+
+The combination of SmallRye JWT with Elytron JWT will be the pair of projects responsible for providing the functionality described in the MicorProfile JWT specification combined with tying in the integration to the application server's security framework to ensure aspects such as identity association and establishment are correctly performed.
+
+Once an identity has been established according to the steps described by the MicroProfile JWT specification the remaining security decisions required are covered by the relevant Jakarta EE specifications in use by the various subsystems. 
+
+=== Nice-to-Have Requirements
+
+It will not be a priority of this enhancement to add support for alternative authentication mechanisms to `MP-JWT` as this brings a lot of additional complexity.  For this initial integration how we provide support for the `MP-JWT` mechanism should be largely transparent to the deployment allowing us to evolve this aspect further.
+
+Some of the issues we need to consider may be worthwhile discussing at the specification level.
+
+Presently SmallRye JWT is making use of EE Security to provide an authentication mechanism.  A big benefit of making use of EE Security is that a deployment can also make use of this specification to provide an `@IdentityStore` implementation as described in the MP JWT specification.  Deployments also have the option of bringing in their own authentication mechanisms if they are not making use of `MP-JWT`.
```

Comment:
+1 to using EE security. I've been using SmallRye JWT on WildFly for several months (bundled in the web archive) and it has worked nicely - currently running 17.0.1. The big benefit to using EE security is the availability of the principal in downstream components, i.e. EJBs.

Reply:
I would suggest the availability of the principal in downstream components is not a benefit that will be gained by using EE Security - once this integration in in place the integration will be establishing a SecurityIdentity using the server's security SPIs so an identity would already be propagated as all of the sub-seployments are sharing a common SecurityDomain anyway.

This isn't going to need to be an immediate decision but the decisions around using EE Security vs Native Security vs ???? I suspect will be more focused on the mechanisms themselves with the following considerations: -
- MP-JWT
- Standard Mechanisms
- Vendor Mechansism
- Concurrent Mechanisms
- User supplied mechanisms (managed)
- User supplied mechanisms (deployed).
- Managed security realms
- Deployed IdentityStores